include/drivers/kscan.h

```diff
@@ -44,6 +50,25 @@ typedef void (*kscan_callback_t)(const struct device *dev, uint32_t row,
 				 uint32_t column,
 				 bool pressed);
 
+/**
+ * @brief Keyboard scan ISR callback called when user presses to trigger the
+ * keybaord scan interrupt.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ */
+typedef void (*kscan_isr_callback_t)(const struct device *dev);
+
+/**
+ * @brief Keyboard matrix configuration.
+ *
+ * This defines keyboard matrix configuration used by keyboard scan task and driver.
+ */
+struct kscan_matrix_config {
+	kscan_callback_t callback;
```

Comment:
done

Reply:
I'm confused about this callback.  This callback is getting set by the `kscan_npcx.c` in it's `config_matrix` routine:

```
static int kscan_npcx_configure_matrix(const struct device *dev, struct kscan_matrix_config *config)
{
	struct kscan_npcx_data *const data = dev->data;

	data->config = config;

	/* Assign enable flag to task configuration */
	config->enable_scan = &data->enable_scan;
	/* Configure callback function between kscan task and driver */
	config->callback = kscan_npcx_matrix_callback;

	return 0;
}
```

So the kscan matrix polling task calls the NPCX driver, which then calls the user's callback.

Can the polling task call the user's callback directly?

Do you anticipate users/applications calling both `kscan_config()` and `kscan_matrix_config()`?  Seems like an application will only use one or the other.

Reply:
>Can the polling task call the user's callback directly?  

The user's callback is registered via the kscan_config(), the polling task has no information of the user's callback.
That why the polling task needs the member callback in the `struct kscan_matrix_config`  to link to kscan_npcx_matrix_callback which call the user's callback.

>Do you anticipate users/applications calling both kscan_config() and kscan_matrix_config()? Seems like an application will only use one or the other.

To test the kscan driver and kscan matrix polling task, we run the [kscan sample code](https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/drivers/kscan) on NPCX EVB. The application (src/main.c) registers its callback to npcx kscan driver via kscan_config().
The polling task get a link to the user's callback via kscan_matrix_config().
After the polling task finishes the scanning and detect a key press/release event, it can use the callback to report the event to the  application.

Reply:
As the code works today, the user registers a callback with the kscan driver, and the driver directly invokes this callback after keypress events.

This PR is effectively adding a new keyboard type - matrix keyboards.

What about changing this up so that the `kscan_matrix` becomes it's own keyboard driver?  The NPCX driver would be a child of the matrix driver. 

This would work as follows:

The `kscan_npcx.c` driver would register itself with the `kscan_matrix` driver, using a new API `kscan_matrix_api`.  `kscan_matrix` then registers itself with the existing `kscan` driver.

This would look like:
```
/* kscan_npcx.c */
static const struct kscan_matrix_api kscan_matrix_ncpx_driver_api = {
	.config_matrix = kscan_npcx_configure_matrix,
	.drive_matrix_column = kscan_npcx_drive_matrix_column,
	.read_matrix_row = kscan_npcx_read_matrix_row,
	.resume_matrix_detection = kscan_npcx_resume_matrix_detection,
}

/* kscan_matrix.c */
static const struct kscan_driver_api kscan_matrix_driver_api = {
	.config = kscan_npcx_configure,
	.enable_callback = kscan_npcx_enable_interface,
	.disable_callback = kscan_npcx_disable_interface,
};
```

The users still call `kscan_config()`, `kscan_disable_callback()`, and `kscan_enable_callback()`.  But these routines call into the `kscan_matrix` driver, and the matrix driver then calls the chip level driver for the matrix operations.

It looks like the ITE and MCHP drivers could be converted to used the `kscan_matrix` driver.

The logical organization would be:

```
kscan drivers:
    cst8186s
    ft5336
    ht16k33
    matrix
        npcx
        ite
        mchip

```

Reply:
Done.